app/src/main/java/org/kiwix/kiwixmobile/ZimContentProvider.java

```diff
@@ -70,29 +71,28 @@
 
   public static String zimFileName;
 
-  @Inject public static JNIKiwix jniKiwix;
+  @Inject
+  public JNIKiwix jniKiwix;
 
   public static JNIKiwixReader currentJNIReader;
 
   public static JNIKiwixSearcher jniSearcher;
 
-  @Inject public static Context context;
-  
   private static ArrayList<String> listedEntries;
 
   public void setupDagger() {
-    KiwixApplication.getInstance().getApplicationComponent().inject(this);
+    KiwixApplication.getApplicationComponent().inject(this);
     setIcuDataDirectory();
     jniSearcher = new JNIKiwixSearcher();
     listedEntries = new ArrayList<>();
   }
 
 
-  private static String getFulltextIndexPath(String file){
+  private static String getFulltextIndexPath(String file) {
     String[] names = {file, file};
 
     /* File might be a ZIM chunk like foobar.zimaa */
-    if (!names[0].substring(names[0].length() - 3).equals("zim")){
+    if (!names[0].substring(names[0].length() - 3).equals("zim")) {
```

Comment:
I know these changes are correct however could they be moved to another commit/not done here. This is already a mammoth task to review.